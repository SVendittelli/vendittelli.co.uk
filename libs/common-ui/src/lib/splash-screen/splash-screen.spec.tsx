import { fireEvent, render, screen } from '@testing-library/react';

import SplashScreen from './splash-screen';

describe('SplashScreen', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <SplashScreen dismissText="Dismiss" onDismiss={() => ({})}>
        Test
      </SplashScreen>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should call dismiss prop when button is clicked', () => {
    const dismiss = jest.fn();
    render(
      <SplashScreen dismissText="Dismiss" onDismiss={dismiss}>
        Test
      </SplashScreen>
    );

    fireEvent.click(screen.getByRole('button'));

    expect(dismiss).toHaveBeenCalled();
  });
});
