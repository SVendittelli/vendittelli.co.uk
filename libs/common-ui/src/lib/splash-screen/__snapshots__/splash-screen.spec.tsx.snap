// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen should render successfully 1`] = `
<body>
  <div>
    <main
      class="container"
    >
      Test
      <button>
        Dismiss
      </button>
    </main>
  </div>
</body>
`;
